src/OrchardCore.Modules/OrchardCore.ContentFields/Views/TextField-Tel.Edit.cshtml

```diff
@@ -0,0 +1,16 @@
+@model OrchardCore.ContentFields.ViewModels.EditTextFieldViewModel
+@using OrchardCore.ContentManagement.Metadata.Models
+@using OrchardCore.ContentFields.Settings;
+
+@{
+    var settings = Model.PartFieldDefinition.Settings.ToObject<TextFieldSettings>();
+}
+
+<fieldset class="form-group">
+    <label asp-for="Text">@Model.PartFieldDefinition.DisplayName()</label>
+    <input asp-for="Text" class="form-control content-preview-text" type="tel" />
```

Comment:
This should also add a pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" attribute to the input else this will render as a simple textbox. Right now this input has no validation ; I could add letters to it though someone could say that this is for supporting phone numbers like "1-800-MY-APPLE"

Reply:
Don't add any validation, or more patterns. It's very locale specific, and leaving the browser do that is fine right now. Otherwise just remove the editor.

Reply:
We could give the option of a pattern and if the pattern is null then don't render one. Else this field is a textbox.

Reply:
Too complex. At that point better teach them how to write a custom editor. There are way too many options to handle phone numbers.

Reply:
Documentation then